community/lucene-index/src/main/java/org/neo4j/index/impl/lucene/legacy/LuceneDataSource.java

```diff
@@ -111,6 +115,45 @@ public String toString()
     private boolean closed;
     private LuceneFilesystemFacade filesystemFacade;
     private IndexReferenceFactory indexReferenceFactory;
+    private Map<IndexIdentifier,Map<String,DocValuesType>> indexTypeMap;
```

Comment:
this map is never cleaned, i would expect that we will remove index information from it when the index is removed